### PR TITLE
Stat auto (Price, Supply, MC)

### PR DESCRIPTION
**Pull request by Shaharyar-saleem**

### CODE REVIEW

File: js/stats.js [hunk 1]
```diff
@@ -1,24 +1,40 @@
+let price = async function () {
+  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
+  let data = await res.json();
+  return data;
+}
```
Comment (bobachef) [inline js/stats.js:5]: shouldn't use `let` but also easier to just name the function.
```
async function getPriceData() {
  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
  let data = await res.json();
  return data;
}
```

File: js/stats.js [hunk 2]
```diff
@@ -1,24 +1,40 @@
+let price = async function () {
+  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
+  let data = await res.json();
```
Comment (bobachef) [inline js/stats.js:3]: res.json doesn't return a promise so no need to await it...

File: js/stats.js [hunk 3]
```diff
@@ -1,24 +1,40 @@
+let price = async function () {
+  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
+  let data = await res.json();
+  return data;
+}
+
+let supply = async function () {
```
Comment (bobachef) [inline js/stats.js:7]: same thing change this to 

```async function getSupplyData()```

File: js/stats.js [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
+  return data;
+}
+
+async function caller(){
+  let _coinPrice, _coinSupply;
+  _coinPrice = await price().then((data)=>{
+    let price;
+      price = parseFloat(data.priceUSD).toFixed(14);
+      return price;
+  })
```
Comment (bobachef) [inline js/stats.js:19]:
```
const priceData = await getPriceData()
let price =  parseFloat(data.priceUSD).toFixed(14);
```

File: js/stats.js [hunk 5]
```diff
@@ -1,24 +1,40 @@
...
+  return data;
+}
+
+let supply = async function () {
+  let res = await fetch("https://api.bscscan.com/api?module=account&action=tokenbalance&contractaddress=0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa&address=0x0000000000000000000000000000000000000001&tag=latest&apikey=428F24U49DPCZXAG41HUV52GREUXXI8ERR")
+  let data = await res.json();
+  return data;
+}
+
+async function caller(){
```
Comment (bobachef) [inline js/stats.js:13]: rename this function to `calculatePriceSupplyAndDisplay`

File: js/stats.js [hunk 6]
```diff
@@ -1,24 +1,40 @@
+let price = async function () {
+  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
+  let data = await res.json();
+  return data;
+}
+
+let supply = async function () {
+  let res = await fetch("https://api.bscscan.com/api?module=account&action=tokenbalance&contractaddress=0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa&address=0x0000000000000000000000000000000000000001&tag=latest&apikey=428F24U49DPCZXAG41HUV52GREUXXI8ERR")
+  let data = await res.json();
```
Comment (bobachef) [inline js/stats.js:9]: remove `await`

File: js/stats.js [hunk 7]
```diff
@@ -1,24 +1,40 @@
...
-      document.getElementById("coin-price").innerText = price;
-    });
-  return coin_price
+  _coinSupply = await supply().then((data)=>{
+    let supply;
+      const burnedSupply = data.result.slice(0,15);
+        supply = 1000000000000000-burnedSupply;
+        return supply;
+  })
+  display(_coinPrice , _coinSupply);
```
Comment (bobachef) [inline js/stats.js:27]:
```
const TOTAL_SUPPLY = 1000000000000000
const supplyData = await getSupplyData()
const burnedSupply = supplyData.result.slice(0,15);
const supply = TOTAL_SUPPLY-burnedSupply;

display(price, supply)

```

File: js/stats.js [hunk 8]
```diff
@@ -1,24 +1,40 @@
...
+  let res = await fetch("https://api.bscscan.com/api?module=account&action=tokenbalance&contractaddress=0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa&address=0x0000000000000000000000000000000000000001&tag=latest&apikey=428F24U49DPCZXAG41HUV52GREUXXI8ERR")
+  let data = await res.json();
+  return data;
+}
+
+async function caller(){
+  let _coinPrice, _coinSupply;
+  _coinPrice = await price().then((data)=>{
+    let price;
+      price = parseFloat(data.priceUSD).toFixed(14);
```
Comment (bobachef) [inline js/stats.js:17]: `let price = parseFloat(data.priceUSD).toFixed(14);`

**Review by bobachef — CHANGES_REQUESTED** [review]
some suggestions

**Comment by bobachef** [comment]
Please resolve the conflicts also

File: js/stats.js (hunk 1 above)
Comment (Shaharyar-saleem) [reply js/stats.js:5]: const getPriceData = async function () {
  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
  let data = res.json();
  return data;
}

Comment (Shaharyar-saleem) [reply js/stats.js:5]: const getPriceData = async function () {
  let res  = await fetch("https://api.dex.guru/v1/tokens/0x3504de9e61fdff2fc70f5cc8a6d1ee493434c1aa-bsc")
  let data = res.json();
  return data;
}

File: js/stats.js (hunk 2 above)
Comment (Shaharyar-saleem) [reply js/stats.js:3]: removed

File: js/stats.js (hunk 7 above)
Comment (Shaharyar-saleem) [reply js/stats.js:27]: followed

File: js/stats.js (hunk 1 above)
Comment (Shaharyar-saleem) [reply js/stats.js:5]: followed

File: js/stats.js (hunk 3 above)
Comment (Shaharyar-saleem) [reply js/stats.js:7]: followed

File: js/stats.js (hunk 5 above)
Comment (Shaharyar-saleem) [reply js/stats.js:13]: followed

File: js/stats.js (hunk 6 above)
Comment (Shaharyar-saleem) [reply js/stats.js:9]: followed

**Review by Shaharyar-saleem — COMMENTED** [review]
reviewed index.html page

**Review by bobachef — APPROVED** [review]
looks good

**Pull request merged**